apps/beeswax/src/beeswax/server/dbms.py

```diff
@@ -88,29 +107,38 @@ def get(user, query_server=None, cluster=None):
         from impala.dbms import ImpalaDbms
         from impala.server import ImpalaServerClient
         DBMS_CACHE[user.id][query_server['server_name']] = ImpalaDbms(
-            HiveServerClientCompatible(ImpalaServerClient(query_server, user)),
-            QueryHistory.SERVER_TYPE[1][0]
+          HiveServerClientCompatible(ImpalaServerClient(query_server, user)),
+          QueryHistory.SERVER_TYPE[1][0]
         )
       elif query_server['server_name'] == 'hms':
         from beeswax.server.hive_metastore_server import HiveMetastoreClient
         DBMS_CACHE[user.id][query_server['server_name']] = HiveServer2Dbms(
-            HiveMetastoreClient(query_server, user),
-            QueryHistory.SERVER_TYPE[1][0]
+          HiveMetastoreClient(query_server, user),
+          QueryHistory.SERVER_TYPE[1][0]
         )
       else:
         from beeswax.server.hive_server2_lib import HiveServerClient
         DBMS_CACHE[user.id][query_server['server_name']] = HiveServer2Dbms(
-            HiveServerClientCompatible(HiveServerClient(query_server, user)),
-            QueryHistory.SERVER_TYPE[1][0]
+          HiveServerClientCompatible(HiveServerClient(query_server, user)),
+          QueryHistory.SERVER_TYPE[1][0]
         )
+    elif RESET_HS2_QUERY_SERVER:
```

Comment:
RESET_HS2_QUERY_SERVER feels a bit brittle. Maybe we should just clear out for all the users at the same time (fine if we assume only 1 HS2 valid at any time) or instead delete the user key when we call reset_ha()?

The main point is that RESET_HS2_QUERY_SERVER is a global var, no per user anyway. And then in this get() we only create new client or reuse the cache (don't manage invalidations)

Reply:
I agree to this but if we clear off all the DBMS cache.. I wonder whether will it affect the functionality of impala session if they are open.  Thats why I was specific to add this functionality in there.  The reason I had to use Global was because to variable localization.  I will check to see if there is another way to doing the same and may be cleanup the code a bit more.

Reply:
Did a check, it was useful just to clear specific user's DBMS cache